ambry-replication/src/main/java/com/github/ambry/replication/ReplicationEngine.java

```diff
@@ -409,6 +452,42 @@ protected List<ReplicaThread> createThreadPool(String datacenter, int numberOfTh
     return replicaThreads;
   }
 
+  /**
+   * Create thread for a remote node
+   * @param dataNodeId the remote node
+   * @param startThread If thread needs to be started when created.
+   * @return {@link ReplicaThread} created for the remote node
+   */
+  private ReplicaThread createThread(DataNodeId dataNodeId, boolean startThread) {
```

Comment:
I think this should solve the issue since its usually a slow host, and not a slow partition on the host.